controllers/clusterresources/awssecuritygroupfirewallrule_controller.go

```diff
@@ -74,6 +74,13 @@ func (r *AWSSecurityGroupFirewallRuleReconciler) Reconcile(ctx context.Context,
 		return models.ReconcileRequeue, err
 	}
 
+	if firewallRule.Annotations[models.ClusterIDAnnotation] == "" {
+		return models.ExitReconcile, nil
+	}
```

Comment:
Let's use something similar to `clusterEvents` as we did for cluster's users instead of entirely relying on annotations

Reply:
agreee, refactored